### PR TITLE
txesbmantool

### PR DESCRIPTION
**Pull request by PLangowski**

### CODE REVIEW

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+git://github.com/Dasharo/coreboot.git;branch=txe_sb_tool;protocol=https \
+"
+PV = "1.0+git${SRCPV}"
+SRCREV = "${AUTOREV}"
+
+LIC_FILES_CHKSUM=" \
+file://${WORKDIR}/git/LICENSEs/GPL-2.0-only.txt;md5=5430828348d2cf7d4b5e8395f774a68e \
+"
+
+S="${WORKDIR}/git/util/txe_sb_tool"
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb:14]:
```suggestion
S = "${WORKDIR}/git/util/txe_sb_tool"
```

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+DESCRIPTION="Utility for generating TXE Secure Boot manifests"
+LICENSE="GPL-2.0-only"
+
+SRC_URI =" \
+git://github.com/Dasharo/coreboot.git;branch=txe_sb_tool;protocol=https \
+"
+PV = "1.0+git${SRCPV}"
+SRCREV = "${AUTOREV}"
+
+LIC_FILES_CHKSUM=" \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb:10]:
```suggestion
LIC_FILES_CHKSUM = " \
```

File: meta-dts-distro/recipes-bsp/packagegroups/packagegroup-txe-secure-boot.bb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+SUMMARY = "txe-secure-boot packagegroup"
+DESCRIPTION = "txe-secure-boot packagegroup"
+
+LICENSE = "MIT"
+
+inherit packagegroup
+
+PACKAGES = " \
+    packagegroup-txe-secure-boot \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/packagegroups/packagegroup-txe-secure-boot.bb:9]: I am not sure if we need packagegroup here. How many more tools do we expect here? I think just one more, and it will not be public.

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb (hunk 1 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb]: 8f766cdf68746821e621fe2abffeee4df10c3e09

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb (hunk 2 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb]: https://github.com/Dasharo/meta-dts/commit/8f766cdf68746821e621fe2abffeee4df10c3e09#diff-96e0ad01b7a312a762805dd7274f4c8c365fd144968bbea421a7dcc14ce57c80R10

File: meta-dts-distro/recipes-bsp/packagegroups/packagegroup-txe-secure-boot.bb (hunk 3 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/packagegroups/packagegroup-txe-secure-boot.bb]: I added it to ` ${PN}-tools-base`
https://github.com/Dasharo/meta-dts/commit/8f766cdf68746821e621fe2abffeee4df10c3e09#diff-b84bac204935ca0a7635a14b31645ba7e78426e334857f91d354f9b4013e1813R40

**Comment by PLangowski** [comment]
@macpijan I'd like to add a cukinia test here, but the tool always returns a non-zero exit code if no actual operation is performed. This would require providing a binary to the platform, which is problematic. There is no `--help` or `--version` option.

**Comment by PLangowski** [comment]
I added a test that checks for the tool's existance d59dce4

**Comment by macpijan** [comment]
> There is no --help or --version option.

You can also leave this feedback in the PR adding this tool. At least having help option would be nice.

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+LICENSE = "GPL-2.0-only"
+
+SRC_URI = " \
+git://github.com/Dasharo/coreboot.git;branch=txe_sb_tool;protocol=https \
+"
+
+SRCREV = "295c158e577400025d01e7ca8ad3beb64e1614a0"
+
+LIC_FILES_CHKSUM = " \
+file://${WORKDIR}/git/LICENSES/GPL-2.0-only.txt;md5=5430828348d2cf7d4b5e8395f774a68e \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb:11]: I guess I will wait for pre-commit rules to land first?

File: meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    export PKG_CONFIG_PATH="${STAGING_LIBDIR}/pkgconfig:${STAGING_DIR_HOST}/usr/lib/pkgconfig"
+}
+
+do_install() {
+    oe_runmake 'DESTDIR=${D}' install
+}
+
+INSANE_SKIP:${PN} += "ldflags"
+
+prefix = "/usr/local"
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb:34]: defintely a bad idea. The smmstoretool PR comments apply here

Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/txesbmantool_git.bb]: Done 28e45f9ffe9e01a4b1e5d45d933ed98f16ff668b

**Review by macpijan — APPROVED** [review]
